### ISSUE TITLE
Bug: String indexer/`query` methods can't handle repeating fields

### ISSUE THREAD

**Issue by wokket** [wokket/rust-hl7#26]
No support for handling multiple repeats (`~` delim by default)

**Comment by sempervictus** [comment]
Will take a look at this - feel free to assign me.

**Comment by sempervictus** [comment]
Could i ask for a test-case which needs to pass here?

**Comment by wokket** [comment]
I'll get you a code sample later (real work atm), but for:
```
OBX|1|NM|301.0500^White Blood Count (WBC)^00065227^6690-2^Leukocytes^pCLOCD|1|10.1|10\S\9/L|3.1-9.7|H||A~S|F|||201411130916|MYFAC^MyFake Hospital^L|
```

I'd expect a query string of `F10.R1` to return _either_ `A` or `S` depending on whether we're 0 or 1-based indexing (I'd have to check the spec for that).

**Comment by sempervictus** [comment]
Thanks, will look into it today

**Comment by sempervictus** [comment]
Proposed fix and added test in the refactor PR - #31.
